lib/identity_cache/query_api.rb

```diff
@@ -183,6 +184,7 @@ def get_embedded_association(record, association, options) #:nodoc:
 
       def coder_from_record(record) #:nodoc:
         unless record.nil?
+          record.readonly! if IdentityCache.fetch_read_only_records && IdentityCache.should_use_cache?
```

Comment:
I don't think we should have to check `IdentityCache.should_use_cache?` here, since I think this method is only used to serialize a record to store it in the cache.